apps/tailwind-components/components/input/Tree.vue

```diff
@@ -0,0 +1,167 @@
+    isMultiSelect?: boolean;
+    mobileDisplay?: boolean;
+    expandSelected?: boolean;
+    isRoot?: boolean;
+    inverted?: boolean;
+  }>(),
+  {
+    isMultiSelect: true,
+    mobileDisplay: false,
+    expandSelected: false,
+    isRoot: true,
+    inverted: false,
+  }
+);
+
+// expand status is internally controlled by the component
+// whereas the selected status is controlled by the parent
+const expandedNodes = ref<string[]>([]);
+
+const emit = defineEmits(["update:modelValue"]);
+
+function toggleExpand(nameName: string) {
+  const index = expandedNodes.value.indexOf(nameName);
+  if (index > -1) {
+    expandedNodes.value.splice(index, 1);
+  } else {
+    expandedNodes.value.push(nameName);
+  }
+}
+
+function expandSelection(node: ITreeNode) {
+  let selection: string[] = [];
+  node.children.forEach((child) => {
+    selection.push(child.name);
+    if (child.children) {
+      selection = [...selection, ...expandSelection(child)];
+    }
+  });
+
+  return selection;
+}
+
+function toggleSelect(node: ITreeNode) {
+  if (props.modelValue.includes(node.name)) {
+    // remove node from selected nodes and emit new model
+    emit(
+      "update:modelValue",
+      props.modelValue.filter((n) => n !== node.name)
+    );
+  } else {
+    let currnetSelection = [...props.modelValue, node.name];
+    if (props.expandSelected) {
+      currnetSelection = [...currnetSelection, ...expandSelection(node)];
+    }
+    const deduplicated = [...new Set(currnetSelection)];
+    // clone model and add new before emitting
+    emit("update:modelValue", deduplicated);
+  }
+}
+
+function handleChildSelect(selected: string[], parent: ITreeNode) {
+  const siblingNames = parent.children.map((n) => n.name);
+  const selectParent =
+    siblingNames.some((siblingName) => selected.includes(siblingName)) &&
+    siblingNames.every((siblingName) => selected.includes(siblingName));
+  const updatedSelection = selectParent ? [...selected, parent.name] : selected;
+
+  const deduplicated = [...new Set(updatedSelection)];
+  emit("update:modelValue", deduplicated);
+}
+</script>
+
+<template>
+  <ul
+    :class="[
+      inverted
+        ? 'text-search-filter-group-title-inverted'
+        : 'text-search-filter-group-title',
+    ]"
+  >
+    <li v-for="node in nodes" :key="node.name" class="mt-2.5 relative">
+      <span class="flex items-center">
+        <span
+          v-if="node.children?.length"
+          @click.stop="toggleExpand(node.name)"
+          class="-left-[11px] top-0 rounded-full hover:cursor-pointer h-6 w-6 flex items-center justify-center absolute z-20"
+          :class="[
+            inverted
+              ? 'text-search-filter-group-toggle-inverted hover:bg-search-filter-group-toggle-inverted'
+              : 'text-search-filter-group-toggle hover:bg-search-filter-group-toggle',
+          ]"
+        >
+          <BaseIcon
+            :name="
+              expandedNodes.includes(node.name) ? 'caret-down' : 'caret-up'
```

Comment:
Perhaps css could be used here instead? I'm wondering what the performance differences are when inserting/removing elements vs rotating the icon

Reply:
yes would probably be faster, think this is also done on other icons , i'll have a look